src/ocl/convolutionocl.cpp

```diff
@@ -588,12 +639,23 @@ static void ConvForwardCheckNumerics(const Handle& handle,
         return;
     }
 
-    miopen::checkNumericsInput(handle, tensors.xDesc, tensors.x);
-    miopen::checkNumericsInput(handle, tensors.wDesc, tensors.w);
+    bool flag = false;
+
+    flag |= miopen::checkNumericsInput(handle, tensors.xDesc, tensors.x);
+    flag |= miopen::checkNumericsInput(handle, tensors.wDesc, tensors.w);
 
     worker();
 
-    miopen::checkNumericsOutput(handle, tensors.yDesc, tensors.y);
+    flag |= miopen::checkNumericsOutput(handle, tensors.yDesc, tensors.y);
+
+    const char* file_name = miopen::GetStringEnv(MIOPEN_DUMP_TENSOR_PATH{});
+    if(flag && static_cast<bool>(file_name))
```

Comment:
This should be coded like this, no static_cast please:
```C++
const auto file_name = miopen::GetStringEnv(MIOPEN_DUMP_TENSOR_PATH{});
if(... && file_name != nullptr)...
```

Reply:
agreed!